<launch>

  <node pkg="vicon"
        name="vicon"
        type="vicon"
        output="screen">
  </node>

  <group ns="Coax56">

    <node pkg="vicon_calibrate"
          name="vicon_calibrate"
          type="vicon_calibrate"
          output="screen"
          clear_params="true">
      <remap from="~values" to="/vicon/values"/>
      <remap from="~names" to="/vicon/names"/>
      <param name="calibration" value="$(find coax_vsk)/calibration/Coax56.xml"/>
    </node>

    <node pkg="vicon2odometry"
          name="vicon2odometry"
          type="vicon2odometry"       
          output="screen">
      <remap from="~names" to="/vicon/names"/>
      <remap from="~values" to="/vicon/values"/>
      <remap from="~pose" to="/Coax56/vicon_calibrate/pose"/>
      <remap from="~odom" to="/Coax56/vicon2odometry/odom"/>
      <param name="vsk" value="$(find coax_vsk)/vsk/Coax56.vsk"/>

    </node>

<!--
    <node pkg="coax_server"
          name="coax"
          type="coax_server"
	  output="screen"
          args="/dev/tty.usbserial-A800f3Zv:1">
      <remap from="/coax_server/state" to="/Coax56/state"/>

      <remap from="/coax_server/control" to="/Coax56/control"/>
      <remap from="/coax_server/rawcontrol" to="/Coax56/rawcontrol"/>

      <remap from="/coax_server/reach_nav_state" to="/Coax56/reach_nav_state"/>
      <remap from="/coax_server/configure_comm" to="/Coax56/configure_comm"/>
      <remap from="/coax_server/set_timeout" to="/Coax56/set_timeout"/>
    </node>
-->

    <node pkg="coax_ros_control"
          name="coax_ros_control"
          type="coax_ros_control"
	  output="screen">
      <remap from="/coax_ros_control/odom" to="/Coax56/vicon2odometry/odom"/>
      <remap from="/coax_ros_control/state" to="/Coax56/state"/>

      <remap from="/coax_ros_control/rawcontrol" to="/Coax56/rawcontrol"/>

      <remap from="/coax_ros_control/reach_nav_state" to="/Coax56/reach_nav_state"/>
      <remap from="/coax_ros_control/configure_comm" to="/Coax56/configure_comm"/>
      <remap from="/coax_ros_control/set_timeout" to="/Coax56/set_timeout"/>

      <remap from="/coax_ros_control/set_control_mode" to="/Coax56/set_control_mode"/>

      <param name="frequency" value="10"/>
      <param name="CoaX" value="56"/>

      <rosparam file="$(find coax_ros_control)/config/coax_parameters.yaml"/>
    </node>

  </group>

</launch>
